### PR TITLE
New test for the omp_calloc routine (device version)

### PR DESCRIPTION
**Pull request by jrreap**
Test for the omp_calloc routine if allocation works on a target device properly.

### CODE REVIEW

File: tests/5.1/allocate/test_target_calloc.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  {
+
+    int *x;
+
+    omp_memspace_handle_t  memspace = omp_default_mem_space;
+    omp_alloctrait_t       traits[0] = {};
+    omp_allocator_handle_t alloc = omp_init_allocator(memspace,0,traits);
+
+    x = (int *)omp_calloc(64, N*sizeof(int), alloc);
+
```
Comment (fel-cab) [inline tests/5.1/allocate/test_target_calloc.c:34]: I think you should add something like:
`if (x == NULL) { OMPVV_ERROR("omp_calloc returned null"); return (1); } `

File: tests/5.1/allocate/test_target_calloc.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+#include <stdint.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int test_omp_target_calloc() {
+
+  int errors = 0;
+
+  #pragma omp target map(tofrom: errors) map(to: N) uses_allocators(omp_default_mem_alloc)
```
Comment (spophale) [inline tests/5.1/allocate/test_target_calloc.c:24]: This is not the way to create an allocator in the target region.
See: https://github.com/SOLLVE/sollve_vv/blob/master/tests/5.0/target/test_target_uses_allocators_default.c

Comment (jrreap) [reply tests/5.1/allocate/test_target_calloc.c:24]: Will polish this up and correctly allocate this

File: tests/5.1/allocate/test_target_calloc.c [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  #pragma omp target map(tofrom: errors) uses_allocators(omp_default_mem_alloc)
+  {
+
+    int *x;
+
+    x = (int *)omp_calloc(64, N*sizeof(int), omp_default_mem_alloc);
+
+    if (x == NULL) { 
+      OMPVV_ERROR("omp_calloc returned null"); 
+      return (1);
```
Comment (tob2) [inline tests/5.1/allocate/test_target_calloc.c:33]: GCC complains here:
```
test_target_calloc.c:33:14: error: invalid branch to/from OpenMP structured block
```
(However, GCC does accept `exit(1);` or `abort();`.)

Likewise with Clang:
```
test_target_calloc.c:35:7: error: cannot return from OpenMP region
```

From the OpenMP spec:
* _structured block_ — For C/C++, an executable statement, possibly compound, with a single entry at the9
top and a single exit at the bottom, or an OpenMP construct. [...]

**Review by spophale — APPROVED** [review]

**Review by fel-cab — APPROVED** [review]

File: tests/5.1/allocate/test_target_calloc.c [hunk 4]
```diff
@@ -0,0 +1,82 @@
+//===------------------------ test_calloc_host.c --------------------------===//
```
Comment (tob2) [inline tests/5.1/allocate/test_target_calloc.c:1]:
```suggestion
//===------------------------ test_target_calloc.c ------------------------===//
```

**Pull request merged**